src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/SecurityTokenProvider.cs

```diff
@@ -58,6 +69,8 @@ public virtual bool SupportsTokenCancellation
         }
 
         // protected methods
+        protected abstract SecurityToken GetTokenCore(TimeSpan timeout);
```

Comment:
Good observations.  I think they should be addressed when I get the IAsyncRequestChannel stuff fully Task-based and we look at how and whether the sync paths use the async paths (like CommunicationObject open and close).  I'm trying to avoid a refactor in this PR and just keep the API's as they are in the full framework.

---
In reply to: [78136608](https://github.com/dotnet/wcf/pull/1515#discussion_r78136608) [](ancestors = 78136608)